### DISCUSSION TITLE
Git não esta "rastreando" minhas modificações

### DISCUSSION THREAD

**Discussion by LuisMatheus-dev** [frontendbr/forum#1872] [Perguntas e Respostas]
Oláa, eu iniciei meu projeto, o git estava identificando normalmente minhas modificações , isto é , eles apareciam na zona de staging , mas após fazer o push para o github (`$ git push -u origin master`) , nenhuma alteração que faço agora  é capturada por ele, quando execulto o `$ git status `o que recebo é:
```
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```
Como devo proceder ?

**Comment by evelew** [comment]
Oi Luis, seus commits não estão aparacendo no github, isso?

Da uma olhada no endereço do repositório que tá configurado no seu projeto.
Veja se seus credenciais estão certas também, só por garantia.

Se não encontrar nenhum problema nas configurações inicia de novo o git no seu projeto.

**Comment by fdaciuk** [comment]
Uma outra coisa que pode ser: confere se não tem algum diretório `.git` em algum diretório mais interno do seu projeto. Se isso aconteceu, você provavelmente criou um repositório git dentro de um projeto que já tinha um repo configurado. Aí as alterações desse diretório mais interno não vão ser "vistas" pelo git do repo mais "externo" :)

**Comment by ninetails** [comment]
Dado a #1866, isso acontece usando o terminal emulator do VSCode e no console do próprio Ubuntu (supondo ele de distro)?

**Comment by deppbrazil** [comment]
@LuisMatheus-dev,

Pq vc usou o parâmetro `-u` junto com o push? 

Isso faz com que vc crie um remoto de upstream, ou seja esse problema seu pode ser de remoto iterativo.

Leia isso pode ser que te ajude.
https://www.jquery-az.com/git-push-command/

**Comment by LuisMatheus-dev** [comment]
> Oi Luis, seus commits não estão aparacendo no github, isso?
> 
> Da uma olhada no endereço do repositório que tá configurado no seu projeto.
> Veja se seus credenciais estão certas também, só por garantia.
> 
> Se não encontrar nenhum problema nas configurações inicia de novo o git no seu projeto.

@evelew Na realidade não estou conseguindo commitar mudanças porque elas não estão sendo localizadas pelo git por isso não consigo atualizar meu repositório tanto local quanto remoto.

**Comment by LuisMatheus-dev** [comment]
> Uma outra coisa que pode ser: confere se não tem algum diretório `.git` em algum diretório mais interno do seu projeto. Se isso aconteceu, você provavelmente criou um repositório git dentro de um projeto que já tinha um repo configurado. Aí as alterações desse diretório mais interno não vão ser "vistas" pelo git do repo mais "externo" :)

@fdaciuk Então dei uma olhada mais não encontrei nenhum outro arquivo .git , vieram sim alguns na instalação das dependências para criar o projeto react e também na instalação do expo , mas exclui ambos antes de dar o push.

**Comment by LuisMatheus-dev** [comment]
> Dado a #1866, isso acontece usando o terminal emulator do VSCode e no console do próprio Ubuntu (supondo ele de distro)?

@ninetails , sim rodando na instância do vscode

**Comment by LuisMatheus-dev** [comment]
> @LuisMatheus-dev,
> 
> Pq vc usou o parâmetro `-u` junto com o push?
> 
> Isso faz com que vc crie um remoto de upstream, ou seja esse problema seu pode ser de remoto iterativo.
> 
> Leia isso pode ser que te ajude.
> https://www.jquery-az.com/git-push-command/

@deppbrazil , li o post que você colocou,  mas não entendi muito bem , quando faço o git push -u ele faz uma referencia entre meu local e remoto certo ? 
Então no caso quando faço git status, ele "olha" para o remoto, e por isso ele consegue identificar ?

**Comment by deppbrazil** [comment]
> > @LuisMatheus-dev,
> > Pq vc usou o parâmetro `-u` junto com o push?
> > Isso faz com que vc crie um remoto de upstream, ou seja esse problema seu pode ser de remoto iterativo.
> > Leia isso pode ser que te ajude.
> > https://www.jquery-az.com/git-push-command/
> 
> @deppbrazil , li o post que você colocou, mas não entendi muito bem , quando faço o git push -u ele faz uma referencia entre meu local e remoto certo ?
> Então no caso quando faço git status, ele "olha" para o remoto, e por isso ele consegue identificar ?

Não, para fazer referência entre seu remoto e local usamos o `git push origin master` isso para branch **master**, caso for para branch **dev** usamos `git push origin develop` ou `git push origin development` _(depende como está nomeada seu branch dev, os 2 exemplos são o mais comuns)_, isso faz com que crie uma referência e no próximo **push** é necessáio apenas `git push` nos respectivos branches,

O `git status` olha para seu local e não remoto, ele ve os arquivos modificados, deletados e adicionados locais, estando ou não no seu staging area _(antes do git add **[working area]** depois do git add _**[staging area]**_)_,

O `git push -u`, o sinalizador **-u** é usado para definir a origem como o remoto upstream em sua configuração git. Conforme você empurra um branch com sucesso ou o atualiza, ele adiciona uma referência **upstream**. Quando você envia o branch local com a opção `git push - u` , o branch local é vinculado ao branch remoto automaticamente.

### Da uma lida nesse [aqui](https://www.datacamp.com/community/tutorials/git-push-pull?utm_source=adwords_ppc&utm_campaignid=1455363063&utm_adgroupid=65083631748&utm_device=c&utm_keyword=&utm_matchtype=b&utm_network=g&utm_adpostion=&utm_creative=278443377092&utm_targetid=dsa-429603003980&utm_loc_interest_ms=&utm_loc_physical_ms=1001743&gclid=Cj0KCQjw-O35BRDVARIsAJU5mQUNvgjd45RjF3-k4Upt_rEmzVmFR9aHYIh4FrLvm6cWbMFIxKc8YFgaAtWdEALw_wcB) acho que vai te ajudar! :smiley:
